school/module.py

```diff
@@ -0,0 +1,1433 @@
+                ).format(abbreviation=program_abbreviation, degree=degree)
+            )
+            return
+
+        subject = Subject.get_by_abbreviation(ctx, subject_abbreviation)
+
+        if not subject:
+            await ctx.reply(
+                _(ctx, "Subject with abbreviation {abbreviation} not found.").format(
+                    abbreviation=subject_abbreviation
+                )
+            )
+            return
+
+        obligation = obligation.upper()
+
+        if obligation not in OBLIGATIONS:
+            message = _(ctx, "Obligation must be:")
+            for obligation in OBLIGATIONS:
+                message += (
+                    "\n**"
+                    + obligation
+                    + "** "
+                    + _(ctx, "for")
+                    + " "
+                    + School._translate_obligation(obligation)
+                )
+
+            await ctx.reply(message)
+
+        relation = SubjectProgram.get(subject, program, year, obligation)
+
+        if not relation:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "This combination of program, degree, subject, year and obligation does not exists.",
+                )
+            )
+            return
+
+        relation.delete()
+
+        await ctx.reply(_(ctx, "Subject and program succesfully unlinked."))
+
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Program {program.abbreviation} ({program.degree}) unlinked with {subject.abbreviation} (year: {year}, obligation: {obligation}).",
+        )
+
+    @staticmethod
+    def _translate_degree(ctx, degree: str) -> str:
+        """Translate degree"""
+        if degree == "Bachelor":
+            return _(ctx, "Bachelor")
+        elif degree == "Masters":
+            return _(ctx, "Masters")
+        elif degree == "Doctoral":
+            return _(ctx, "Doctoral")
+        else:
+            return "-"
+
+    @staticmethod
+    def _translate_semester(ctx, semester: str) -> str:
+        """Translate semester"""
+        if semester == "Winter":
+            return _(ctx, "Winter")
+        elif semester == "Summer":
+            return _(ctx, "Summer")
+        elif semester == "Both":
+            return _(ctx, "Both")
+        else:
+            return "-"
+
+    def _translate_obligation(ctx, obligation: str) -> str:
+        """Translate obligation from it's czech shortcut"""
+        if obligation == "V":
+            return _(ctx, "Optional")
+        elif obligation == "PV":
+            return _(ctx, "Compulsory - optional")
+        elif obligation == "P":
+            return _(ctx, "Compulsory")
+        elif obligation == "O":
+            return _(ctx, "Other")
+        elif obligation == "D":
+            return _(ctx, "Recommended")
+        elif obligation == "VO":
+            return _(ctx, "Optional - specialized")
+        else:
+            return "-"
```

Comment:
And this is pretty dumb honestly. Just do the obligations right the first time and add the full string to the DB using an Enum and you don't have to do any magic like this

Reply:
Yes and no - we still must use the translation function with string to make popie aware of those strings.